### DISCUSSION TITLE
Mão na Massa Orientação a objetos: Porque não posso imprimir no console um objeto?

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#512] [Tira Dúvidas] [answered]
Fiz todos os objetos, cursos, artigos, items da carreira e carreira quando pedi pra imprimir no console, ele simplesmente imprimiu os namespaces até chegar na classe Career.

![image](https://github.com/balta-io/forum/assets/43733159/1ab20e2e-bda3-40a4-91e0-125b0f411c4e)

**Comment by henriqueweege** [comment]
Oi, @mlasa! Tudo certo?
Quando interpolamos um objeto, o .NET "automagicamente" chama o método ".ToString()", e este por sua vez retorna o "nome completo" da classe, i.e., namespace + nome da classe.
Existem algumas maneiras de contornar esse comportamento. A primeira delas está na linha 34 do teu código, aonde estão sendo acessadas propriedades do artigo e por isso você verá o valor que essas propriedades contém. A segunda delas é através da sobrescrita do método  ".ToString()", conforme citado [nesse artigo do Balta](https://balta.io/blog/csharp-implicit-operators), e [na doc da Microsoft](https://learn.microsoft.com/pt-br/dotnet/csharp/programming-guide/classes-and-structs/how-to-override-the-tostring-method).

Para te dar um exemplo dessa sobrescrita: no caso da linha 34 dentro da classe "Article" você adicionaria o código abaixo, e na linha 34 apenas precisaria chamar o "Console.WriteLine(article)" que a mensagem seria a mesma.

`
public override string ToString()
    {
        return $"Artigo: {Title} em / {Url}";
    }
`

Valeuuu

**Reply by oimarcella** [reply]
Me deu uma luz, sua resposta, obrigada!

Ainda sim, tenho uma dúvida, eu sobrescrevendo o método ToString, só conseguiria imprimir no console as propriedades definidas dentro do método, certo?

Eu não conseguiria fazer algo, como no javascript, que eu imprimo um objeto, e ele me mostra da seguinte maneira: 
```
//qualquer comando para imprimir no console
Console.Write(article);

//output
{
title: "Artigo 1",
order: 2,
url:"artigo-1"
}
```

**Reply by andrebaltieri** [reply, marked as answer]
Sim, conseguiria, mas teria que converter para JSON seu objeto antes do retorno.

Algo assim:
```csharp
public override string ToString()
{
   return JsonSerializer.Serialize(this);
}
```

**Reply by oimarcella** [reply]
Huuum interessante! Obrigada
